addons/website_slides/static/src/js/slides_course_page.js

```diff
@@ -0,0 +1,148 @@
+/** @odoo-module **/
+
+import publicWidget from 'web.public.widget';
+import session from 'web.session';
+import { qweb as QWeb } from 'web.core';
+
+
+/**
+ * Global widget for both fullscreen view and non-fullscreen view of a slide course.
+ * Contains general methods to update the UI elements (progress bar, sidebar...) as well
+ * as method to mark the slide as completed / uncompleted.
+ */
+export const SlideCoursePage = publicWidget.Widget.extend({
+    events: {
+        'click button.o_wslides_button_complete': '_onClickComplete',
+    },
+
+    custom_events: {
+        'slide_completed': '_onSlideCompleted',
+        'slide_to_complete': '_onSlideToComplete',
+    },
+
+    xmlDependencies: ['/website_slides/static/src/xml/website_slides_sidebar.xml'],
+
+    /**
+     * Greens up the bullet when the slide is completed
+     *
+     * @public
+     * @param {Integer} slideId
+     * @param {Boolean} completed
+     */
+    setSlideCompleted: function (slide, completed = true) {
```

Comment:
A bit confusing to have "setSlideCompleted" and "_setCompleted".

"toggleCompletionButton" and "_toggleSlideCompleted" maybe?